### PR TITLE
use material TextInputLayout for edit waypoint page (rel. to #10763)

### PR DESCRIPTION
**Pull request by moving-bits**
## Description
- use material TextInputLayout for edit waypoint page 
- replace old viewbindings (colliding with Gradle plugin 5) with ViewBinding class

|default|multi-line edit|with bearing + distance|
|---|---|---|
|image|image|image|

(First two screenshots are from editing a given waypoint, third screenshot is from creating a user-defined waypoint)

## Additional context
- use style `textinput_dropdown` if you are embedding a single line `AutoCompleteTextView`
- use style `textinput_edittext` if you are embedding a single or multi line `EditText`
- use style `textinput_embedded` for the embedded `AutoCompleteTextView` or `EditText`

## Note
If you want to set visibility to an embedded `AutoCompleteTextView` or `EditText`, you will have to set the visibility of the surrounding `TextInputLayout`, otherwise it will not work.

### CODE REVIEW

File: main/res/layout/cache_filter_generic_string.xml [hunk 1]
```diff
@@ -54,8 +54,6 @@
     <com.google.android.material.textfield.TextInputLayout
         android:id="@+id/searchtext"
         android:hint="@string/cache_filter_stringfilter_searchtext_hint"
-        android:layout_width="match_parent"
-        android:layout_height="wrap_content"
         style="@style/textinput_dropdown"
         app:endIconMode="clear_text"
```
Comment (eddiemuc) [inline main/res/layout/cache_filter_generic_string.xml:58]: Is it possible to add "app:endIconMode="clear_text" also to the "textinput_*" style?

File: main/res/layout/cache_filter_generic_string.xml [hunk 2]
```diff
@@ -54,8 +54,6 @@
     <com.google.android.material.textfield.TextInputLayout
         android:id="@+id/searchtext"
         android:hint="@string/cache_filter_stringfilter_searchtext_hint"
-        android:layout_width="match_parent"
-        android:layout_height="wrap_content"
         style="@style/textinput_dropdown"
```
Comment (eddiemuc) [inline main/res/layout/cache_filter_generic_string.xml:57]: Isn't it necessary to add the new style "textinput_embedded" also to the enclosed "Autocompletetextview" here?

File: main/res/values/styles.xml [hunk 3]
```diff
@@ -198,14 +198,26 @@
         <item name="android:autoText">true</item>
```
Comment (eddiemuc) [inline main/res/values/styles.xml:198]: "textinput_dropdown" seems to be the same as "textinput_edittext_singleline" currently. I support the idea to keep the separate style for the future, but couldn't it just be left empty with parent="textinput_edittext_singleline" for now?

**Review by eddiemuc — APPROVED** [review]

**Comment by eddiemuc** [comment]
@moving-bits thanks for the integration of generic styles for material.io text inputs! I vote for a quick merge so we can build upon it.

File: main/res/layout/cache_filter_generic_string.xml (hunk 2 above)
Comment (moving-bits) [reply main/res/layout/cache_filter_generic_string.xml:57]: Yes, it should - seems I have forgotten it - too many iterations yesterday... Will add it tonight.

File: main/res/layout/cache_filter_generic_string.xml (hunk 1 above)
Comment (moving-bits) [reply main/res/layout/cache_filter_generic_string.xml:58]: I have tried that, but the compiler complained about not knowin "app:endIconMode" attribute in `styles.xml`, so I keept it in the layout files.

File: main/res/values/styles.xml (hunk 3 above)
Comment (moving-bits) [reply main/res/values/styles.xml:198]: Inheritance is a bit different: `Widget.MaterialComponents.TextInputLayout.OutlinedBox.ExposedDropdownMenu` is parent for `textinput_dropdown`, whereas `Widget.MaterialComponents.TextInputLayout.OutlinedBox` (without `ExposedDropdownMenu`) is parent for `textinput_edittext`. 
I had tried using `textinput_dropdown` for the `EditText` definitions as well first, but that did not work for me, I did not get the colored border, so I started searching for other style templates and ended up with that one.

File: main/res/layout/cache_filter_generic_string.xml (hunk 1 above)
Comment (eddiemuc) [reply main/res/layout/cache_filter_generic_string.xml:58]: I googled this quickly and found this: https://stackoverflow.com/questions/39336054/using-app-namespace-in-style
It seems that for "app" attributes in styles, you simply leave out the "app:" praefix

File: main/res/values/styles.xml (hunk 3 above)
Comment (eddiemuc) [reply main/res/values/styles.xml:198]: You're right, I have overlooked that. Sorry, should have known...

**Comment by moving-bits** [comment]
> @moving-bits thanks for the integration of generic styles for material.io text inputs! I vote for a quick merge so we can build upon it.

Thanks for taking time to review. I'll try to fix the remaining issues later today, and then we can merge.

File: main/res/layout/cache_filter_generic_string.xml (hunk 2 above)
Comment (moving-bits) [reply main/res/layout/cache_filter_generic_string.xml:56]: done

File: main/res/layout/cache_filter_generic_string.xml (hunk 1 above)
Comment (moving-bits) [reply main/res/layout/cache_filter_generic_string.xml]: Yeah, should have known that. Have used that many times in the past, but it did not come into my mind yesterday... :-(

Adjusted the styles accordingly.

**Pull request merged**